laws/src/main/scala/cats/laws/discipline/DecideableTests.scala

```diff
@@ -0,0 +1,84 @@
+package cats
+package laws
+package discipline
+
+import org.scalacheck.{Arbitrary, Cogen}
+import org.scalacheck.Prop._
+
+trait DecideableTests[F[_]] extends ContravariantMonoidalTests[F] {
+  def laws: DecideableLaws[F]
+
+  def decideable[A: Arbitrary, B: Arbitrary, C: Arbitrary](implicit
+                                                           arbFA: Arbitrary[F[A]],
+                                                           arbFB: Arbitrary[F[B]],
+                                                           arbFC: Arbitrary[F[C]],
+                                                           CogenA: Cogen[A],
+                                                           CogenB: Cogen[B],
+                                                           CogenC: Cogen[C],
+                                                           EqFA: Eq[F[A]],
+                                                           EqFB: Eq[F[B]],
+                                                           EqFC: Eq[F[C]],
+                                                           EqFABC: Eq[F[(A, B, C)]],
+                                                           EqCC: Eq[F[Either[C, C]]],
+                                                           EqFEitABC: Eq[F[Either[Either[A, B], C]]],
+                                                           iso: SemigroupalTests.Isomorphisms[F],
+                                                           iso2: DecideableTests.Isomorphisms[F]): RuleSet =
+    new RuleSet {
+      val name = "decideable"
+      val parents = Seq(contravariantMonoidal[A, B, C])
+      val bases = Seq.empty
+      val props = Seq(
+        "decideable right absorption" ->
+          forAll(laws.decideableDecideRightAbsorption[A] _),
+        "decideable sum associativity" -> forAll(
+          (fa: F[A], fb: F[B], fc: F[C]) => iso2.associativity(laws.decideableSumAssociativity[A, B, C](fa, fb, fc))
+        ),
+        "decideable right distributivity" ->
+          forAll(laws.decideableRightDistributivity[A, B, C] _)
+      )
+    }
+}
+
+object DecideableTests {
+  def apply[F[_]: Decideable](implicit ev: Isomorphisms[F]): DecideableTests[F] =
+    new DecideableTests[F] { def laws: DecideableLaws[F] = DecideableLaws[F] }
+
+  trait Isomorphisms[F[_]] {
```

Comment:
Do we really need these `Isomorphisms`? 
`Decidable` extends `Contravariant` so we can just that instance, no?

For `Semigroupal` which doesn't have access to an `Invariant` instance this sort of thing is needed to test the laws, but here it shouldn't be needed IMO :)